superset/security/manager.py

```diff
@@ -1702,6 +1702,22 @@ def on_permission_view_after_delete(
         :param target: The mapped instance being persisted
         """
 
+    @staticmethod
+    def get_exclude_users_from_lists() -> List[str]:
+        """
+        Override to dynamically identify a list of usernames to exclude from
+        all UI dropdown lists, owners, created_by filters etc...
+
+        It will exclude all users from the all endpoints of the form
+        ``/api/v1/<modelview>/related/<column>``
+
+        Optionally you can also exclude them using the `EXCLUDE_USERS_FROM_LISTS`
+        config setting.
+
+        :return: A list of usernames
+        """
+        return []
```

Comment:
Should this be a list of ids instead of usernames? Internal filtering should preferably use IDs as they are more determinant.

Reply:
`username` is unique so it has the same level of determinism has `id`, I can't internally identify service users that are dynamically created by `id`.

Reply:
A username can be changed, but id cannot right? By definition that makes `id` more deterministic. It may not happen often in this case, but I've seen things broken in other systems because a username was changed.

Wouldn't you be able to get the user id by just going to the users CRUD views after they are created? The user id is in the `/users/edit/:userId` URL.

Reply:
I could, but why make an extra query to fetch an ID? 
Service users normally have pre-defined usernames, does not seem common practice to me to identify users on a config key by user IDs. There are lot's of cases of pre-defined usernames for service users or special well known users: postgres, root, rdsadmin, rdsrepladmin, operator, backup, guest, anonymous, QSECOFR. We just assume these names will not change.